src/main/java/com/gmail/nossr50/util/blockmeta/chunkmeta/PrimitiveChunkStore.java

```diff
@@ -67,25 +71,32 @@ public void setDirty(boolean dirty) {
     }
 
     @Override
+    @Contract(pure = true)
     public int getChunkX() {
         return cx;
     }
 
     @Override
+    @Contract(pure = true)
     public int getChunkZ() {
         return cz;
     }
 
     private boolean isInvalid(int x, int y, int z) {
         if (y < 0 || y >= worldHeight) return true;
-        return ((x | z) & ~0xF) != 0;
+        return ((x | z) & ~0xF) != 0; // < 0 || > 15
     }
 
     @Override
+    @Contract(pure = true)
     public boolean isTrue(int x, int y, int z) {
         if (isInvalid(x, y, z)) return false;
+        // An explanation of the operation of shift is on lines 26 to 42
```

Comment:
Line numbers change when the file is edited, I'd prefer a comment that didn't need to be regularly updated